pkg/chrootarchive/archive_unix.go

```diff
@@ -110,7 +110,9 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 		// when `xz -d -c -q | storage-untar ...` failed on storage-untar side,
 		// we need to exhaust `xz`'s output, otherwise the `xz` side will be
 		// pending on write pipe forever
-		io.Copy(io.Discard, decompressedArchive)
+		if _, err := io.Copy(io.Discard, decompressedArchive); err != nil {
+			fatal(err)
```

Comment:
AFAICS `fatal` exists to be used in child processes; this is running in the parent and it would terminate the whole process.